pkg/collector/corechecks/cluster/ksm/kubernetes_state.go

```diff
@@ -472,6 +473,15 @@ func (k *KSMCheck) Run() error {
 		return err
 	}
 
+	// Normally the sender is kept for the lifetime of the check.
+	// But as `SetCheckCustomTags` is cheap and `k.instance.Tags` is immutable
+	// It's fast and safe to set it after we get the sender.
+	sender.SetCheckCustomTags(k.instance.Tags)
```

Comment:
I think we should be careful because `CommonConfigure` also calls this function:
https://github.com/DataDog/datadog-agent/blob/83901b1e0543770e83beb1688c18086d3b412e7e/pkg/collector/corechecks/checkbase.go#L142

Which modifies the checkTags
https://github.com/DataDog/datadog-agent/blob/cd791dfffc14f1c05780e5339748352417afdcdd/pkg/aggregator/sender.go#L139